src/runtime/transition/index.ts

```diff
@@ -98,37 +98,40 @@ export interface SlideParams {
 	delay?: number;
 	duration?: number;
 	easing?: EasingFunction;
+	axis?: 'x' | 'y';
 }
 
 export function slide(node: Element, {
 	delay = 0,
 	duration = 400,
-	easing = cubicOut
+	easing = cubicOut,
+	axis = 'y'
 }: SlideParams = {}): TransitionConfig {
 	const style = getComputedStyle(node);
 	const opacity = +style.opacity;
-	const height = parseFloat(style.height);
-	const padding_top = parseFloat(style.paddingTop);
-	const padding_bottom = parseFloat(style.paddingBottom);
-	const margin_top = parseFloat(style.marginTop);
-	const margin_bottom = parseFloat(style.marginBottom);
-	const border_top_width = parseFloat(style.borderTopWidth);
-	const border_bottom_width = parseFloat(style.borderBottomWidth);
-
+	const primary_dimension = axis === 'y' ? 'height' : 'width';
+	const primary_dimension_value = parseFloat(style[primary_dimension]);
+	const secondary_dimensions = axis === 'y' ? ['Top', 'Bottom'] : ['Left', 'Right'];
+	const padding_first_value = parseFloat(style.padding[secondary_dimensions[0]]);
+	const padding_second_value = parseFloat(style.padding[secondary_dimensions[1]]);
+	const margin_first_value = parseFloat(style.margin[secondary_dimensions[0]]);
+	const margin_second_value = parseFloat(style.margin[secondary_dimensions[1]]);
```

Comment:
```suggestion
	const padding_first_value = parseFloat(style.padding + secondary_dimensions[0]);
	const padding_second_value = parseFloat(style.padding + secondary_dimensions[1]);
	const margin_first_value = parseFloat(style.margin + secondary_dimensions[0]);
	const margin_second_value = parseFloat(style.margin + secondary_dimensions[1]);
```

try adding margin or padding to your `.contact-container` element, and you'll notice a jarring jump at the end of the transition, because you didnt' transition the padding / margin correctly, because you didnt read out the padding / margin values correctly!